src/components/Clipboard.react.js

```diff
@@ -0,0 +1,109 @@
+import React, {Component} from 'react'; // eslint-disable-line no-unused-vars
+import PropTypes from 'prop-types';
+import {FontAwesomeIcon} from '@fortawesome/react-fontawesome';
+import {faCopy, faCheckCircle} from '@fortawesome/free-regular-svg-icons';
+
+/**
+ * The Clipboard component copies text to the clipboard
+ */
+
+export default class Clipboard extends React.Component {
+    constructor(props) {
+        super(props);
+        this.copyToClipboard = this.copyToClipboard.bind(this);
+        this.copySuccess = this.copySuccess.bind(this);
+        this.getText = this.getText.bind(this);
+        this.state = {
+            copied: false,
+            hasNavigator: true,
+        };
+    }
+
+    copySuccess() {
+        const showCopiedIcon = 1000;
+        this.setState({copied: true}, () => {
+            setTimeout(() => {
+                this.setState({copied: false});
+            }, showCopiedIcon);
+        });
+    }
+
+    getText() {
+        // get the inner text.  If none, use the content of the value param
+        var text = document.getElementById(this.props.target_id).innerText;
+        if (text === undefined || text === '') {
+            text = document.getElementById(this.props.target_id).value;
+            text = text === undefined ? '' : text;
+        }
+        return text;
+    }
+
+    copyToClipboard() {
+        const text = this.getText();
+        navigator.clipboard.writeText(text).then(
+            () => {
+                this.copySuccess();
+            },
+            function() {
+                alert('copy error');
+            }
+        );
+    }
+
+    componentDidMount() {
+        if (!navigator.clipboard) {
+            this.setState({hasNavigator: false});
+            alert('Copy to clipboard not available with this browser');
+        }
+    }
+
+    render() {
+        const {id, title, className, style} = this.props;
+        const copyIcon = <FontAwesomeIcon icon={faCopy} />;
+        const copiedIcon = <FontAwesomeIcon icon={faCheckCircle} />;
+        const btnIcon = this.state.copied ? copiedIcon : copyIcon;
+
+        return this.state.hasNavigator ? (
+            <div id={id} title={title} style={style} className={className}>
+                <i onClick={() => this.copyToClipboard()}>{btnIcon}</i>
+            </div>
+        ) : null;
+    }
+}
+
+Clipboard.defaultProps = {};
+
+Clipboard.propTypes = {
+    /**
+     * The ID used to identify this component.
+     */
+    id: PropTypes.string,
+
+    /**
+     * id of target component containing text to copy to the clipboard.
+     *  The inner text of the `children` prop will be copied to the clipboard.  If none, then the text from the
+     *  `value` prop will be copied.
+     */
+    target_id: PropTypes.string.isRequired,
```

Comment:
Two thoughts about this:
- There may be use cases where the text you want copied doesn't map cleanly to the contents of a specific DOM element. What if we allowed you to omit `target_id`, and if you do that we fall back on another prop (perhaps just `text`) that could be set by a callback if necessary?
- pattern-matching callbacks use object IDs, which are then stringified in a specific way when rendered into the DOM. Ideally we should do that for the user (in `getText`), so they can pass the object in as `target_id` without having to know and replicate how we stringify. The code we use for that is [here](https://github.com/plotly/dash/blob/19917d8760940ec59acc8224f92982d49566d862/dash-renderer/src/actions/dependencies.js#L114-L123) and it's entirely self-contained and expected to be stable, unless @Marc-Andre-Rivet has a clever idea how to share that code I'd probably just duplicate it here.

Reply:
Regarding the first thought,  I added a `text` prop as you suggested and also  `n_clicks` . 

Also, if there is no  `target_id` or `text`, the copy icon will just act like a button.  This makes it possible to use other copy to clipboard methods in a callback like `pandas.DataFrame.to_clipboard` which works great with the DataTable or figure hover data


On the second point, the stringifyID function worked like a charm  :-)